packages/jsii-pacmak/lib/targets/dotnet.ts

```diff
@@ -564,23 +697,24 @@ class DotNetGenerator extends Generator {
 
         this.openFileIfNeeded(name, this.assembly.targets!.dotnet!.namespace, this.isNested((ifc)));
 
+        this.emitDocs(ifc);
         const suffix = `: ${nameutils.convertInterfaceName(ifc.name)}`;
 
         const jsiiAttribute = `[JsiiByValue]`;
         this.code.line(jsiiAttribute);
         this.code.openBlock(`public class ${name} ${suffix}`);
 
         this.flagFirstMemberWritten(false);
-        const emitInstanceGetter = false;
         const datatype = true;
-        this.emitInterfaceMembersForProxyOrDatatype(ifc, emitInstanceGetter, datatype);
+        const proxy = false;
+        this.emitInterfaceMembersForProxyOrDatatype(ifc, datatype, proxy);
         this.code.closeBlock();
         this.closeFileIfNeeded(name, this.isNested(ifc));
     }
 
     // This generates the body of the interface proxy or data type class
     // This loops through all the member and generates them
-    private emitInterfaceMembersForProxyOrDatatype(ifc: spec.InterfaceType | spec.ClassType, emitInstanceGetter: boolean, datatype: boolean): void {
+    private emitInterfaceMembersForProxyOrDatatype(ifc: spec.InterfaceType | spec.ClassType, datatype: boolean, proxy: boolean): void {
```

Comment:
Are datatype and proxy flags mutually exclusive? If so, instead of using booleans, can we use an enum?

Reply:
I taught about it, but chose not to because while they are mutually exclusive in the context of emitInterfaceMembersForProxyOrDatatype() these parameters are not in the context of emitProperty() (They can both be false if we are just emitting a property for a class).